omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -0,0 +1,1898 @@
+
+                # We can pre-grasp in sticky grasping mode only for opening
+                if should_open:
+                    yield from self._execute_grasp()
+
+                # Since the grasp pose is slightly off the object, we want to move towards the object, around 5cm.
+                # It's okay if we can't go all the way because we run into the object.
+                yield from self._navigate_if_needed(obj, pose_on_obj=approach_pose)  #, check_joint=check_joint)
+                
+                if should_open:
+                    yield from self._move_hand_direct_cartesian(approach_pose, ignore_failure=False, stop_on_contact=True, stop_if_stuck=True)
+                else:
+                    print("go to approach pose")
+                    yield from self._move_hand_direct_cartesian(approach_pose, ignore_failure=False, stop_if_stuck=True)
+            
+                # Step once to update
+                empty_action = self._empty_action()
+                yield self.with_context(empty_action)
+
+                for i, target_pose in enumerate(target_poses):
+                    print(f"go to target pose {i}")
+                    yield from self._move_hand_direct_cartesian(target_pose, ignore_failure=False, stop_if_stuck=True)
+
+                # Moving to target pose often fails. Let's get the hand to apply the correct actions for its current pos
+                # This prevents the hand from jerking into its desired position when we do a release.
+                yield from self._move_hand_direct_cartesian(
+                    self.robot.eef_links[self.arm].get_position_orientation(), 
+                    ignore_failure=True,
+                    stop_if_stuck=True
+                )
+
+                if should_open:
+                    yield from self._execute_release()
+                    yield from self._move_base_backward()
+
+            except ActionPrimitiveError as e:
+                print(e)
+                if should_open:
+                    yield from self._execute_release() 
+                    yield from self._move_base_backward()
+                else:
+                    yield from self._move_hand_backward()
+
+        if obj.states[object_states.Open].get_value() != should_open:
+            raise ActionPrimitiveError(
+                ActionPrimitiveError.Reason.POST_CONDITION_ERROR,
+                "Despite executing the planned trajectory, the object did not open or close as expected. Maybe try again",
+                {"target object": obj.name, "is it currently open": obj.states[object_states.Open].get_value()},
+            )
+        
+        # Reset object to track
+        self._tracking_object_pose = None
+        self._track_eef = False
+
+
+    def _move_base_backward(self, steps=5, speed=0.2):
+        """
+        Yields action for the robot to move base so the eef is in the target pose using the planner
```

Comment:
huh? this just moves the base backwards right, irrespective of the EEF?

Reply:
Yup. I don't know why these are implemented this way, and we'll definitely make a v2 of the open/close stuff but need to move on for now.